pymatgen/entries/compatibility.py

```diff
@@ -239,84 +274,275 @@ def get_correction(self, entry) -> float:
+                        ufloat(self.comp_correction[gas], self.comp_errors[gas])
+                        * comp[gas]
+                    )
+
+        return correction
+
+    def __str__(self):
+        return "{} Composition Correction".format(self.name)
+
+
 @cached_class
 class AqueousCorrection(Correction):
     """
     This class implements aqueous phase compound corrections for elements
     and H2O.
     """
 
-    def __init__(self, config_file):
+    def __init__(self, config_file, error_file=None):
         """
         Args:
             config_file: Path to the selected compatibility.yaml config file.
+            error_file: Path to the selected compatibilityErrors.yaml config file.
         """
         c = loadfn(config_file)
-        self.cpd_energies = c['AqueousCompoundEnergies']
+        self.cpd_energies = c["AqueousCompoundEnergies"]
+        # there will either be a CompositionCorrections OR an OxideCorrections key,
+        # but not both, depending on the compatibility scheme we are using.
+        # TODO - the two lines below are specific to MaterialsProjectCompatibility
+        # and MaterialsProjectCompatibility2020. Could be changed to be more general
+        # and/or streamlined if MaterialsProjectCompatibility is retired.
+        self.comp_correction = c.get("CompositionCorrections", defaultdict(float))
+        self.oxide_correction = c.get("OxideCorrections", defaultdict(float))
         self.name = c["Name"]
+        if error_file:
+            e = loadfn(error_file)
+            self.cpd_errors = e.get("AqueousCompoundEnergies", defaultdict(float))
+        else:
+            self.cpd_errors = defaultdict(float)
 
-    def get_correction(self, entry) -> float:
+    def get_correction(self, entry) -> ufloat:
         """
         :param entry: A ComputedEntry/ComputedStructureEntry
-        :return: Correction.
+        :return: Correction, Uncertainty.
         """
+        from pymatgen.analysis.pourbaix_diagram import MU_H2O
         comp = entry.composition
         rform = comp.reduced_formula
         cpdenergies = self.cpd_energies
-        correction = 0
+        correction = ufloat(0.0, 0.0)
         if rform in cpdenergies:
             if rform in ["H2", "H2O"]:
-                correction = cpdenergies[rform] * comp.num_atoms \
-                             - entry.uncorrected_energy - entry.correction
+                corr = (
+                    cpdenergies[rform] * comp.num_atoms
+                    - entry.uncorrected_energy
+                    - entry.correction
+                )
+                err = self.cpd_errors[rform] * comp.num_atoms
+
+                correction += ufloat(corr, err)
             else:
-                correction += cpdenergies[rform] * comp.num_atoms
+                corr = cpdenergies[rform] * comp.num_atoms
+                err = self.cpd_errors[rform] * comp.num_atoms
+
+                correction += ufloat(corr, err)
         if not rform == "H2O":
```

Comment:
@shyamd @montoyjh does this modification to the `AqueousCorrection` make sense to you in light of our discussions?

Reply:
Yeah, this seems to be the closest to what was there before. We'll fix the whole water detection issue later on as that is much harder.